tests/Core/Content/Product/SalesChannel/Listing/ProductListingRouteTest.php

```diff
@@ -55,4 +55,16 @@ public function testWillUseOriginalInCaseTheCategoryIdIsTheMainCategory(): void
         $this->getOriginal()->expects($this->once())->method('load');
         $this->call($productRoute, $request, $salesChannelContextMock, $expectedMainCategoryId);
     }
+
+    public function testWillUseOriginalInCaseRequestComesFromHomepage(): void
```

Comment:
What exactly is Original? Shopware Loader?

Reply:
Yes there are already tests in this class that test the same behavior.